### PR TITLE
Support building and running code samples on Windows

### PR DESCRIPTION
**Pull request by miyuki**
This patch series tidies up documentation and Makefiles of the code samples and adds support for building and running the samples on Windows.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 * newlib
 
-Currently, only Armv6-M can be targeted, and only one library variant is provided targeting the Armv6-M architecture without any optional features.
+Currently, only Armv6-M can be targeted, and only one library variant is
+provided targeting the Armv6-M architecture without any optional features.
 
 ## Goal
 
-The goal is to provide a LLVM based bare-metal toolchain that can target the Arm architecture family from Armv6-M and newer. The toolchain follows the ABI for the Arm Architectue and attempts to provide typical features needed for embedded and realtime operating systems.
+The goal is to provide a LLVM based bare-metal toolchain that can target the
```
Comment (vhscampos) [inline README.md:17]: My Google search told that it should be "a**n** LLVM". But please double check as well.

File: samples/README.md [hunk 2]
```diff
@@ -1,8 +1,53 @@
...
+
+To debug the samples you will need to install GDB provided by the
+[GNU Arm Embedded Toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads).
+
+## Compiling, running and debugging the samples
+
+Change to the directory of a specific sample (e.g. ``src/baremetal-uart``) and
+use the following commands to build, run or debug the sample:
+* `VERSION=0.1 make build` to build the sample.
+* `VERSION=0.1 make run` to run the sample with QEMU emulator
```
Comment (vhscampos) [inline samples/README.md]: Missing period at the end

File: samples/README.md [hunk 3]
```diff
@@ -24,24 +31,31 @@ on Ubuntu Linux it can be installed using the following command:
...
-### Specifying the location of the installed toolchain
+## Specifying the location of the installed toolchain
 
 The Makefiles of the code samples need the location of the installed LLVM
-Embedded Toolchain for Arm. There are several ways to specify the location:
+Embedded Toolchain for Arm. There are several ways to specify the location,
+depending on the environment. The following option works in all environments:
 * Set the environment variable ``BIN_PATH`` to point to the ``bin`` directory
-  of the toolchain, or
+  of the toolchain
```
Comment (vhscampos) [inline samples/README.md:56]: Missing period at the end

File: samples/README.md [hunk 4]
```diff
@@ -76,3 +93,16 @@ syntax:
 ```
 $ VERSION=HEAD make run
 ```
+
+### Windows
+
+If you plan to run the compiled samples ensure that the path of the directory
```
Comment (vhscampos) [inline samples/README.md:99]: Please insert a comma between "samples" and "ensure"

**Review by vhscampos — CHANGES_REQUESTED** [review]

File: samples/src/baremetal-semihosting/README.md [hunk 5]
```diff
@@ -1,21 +1,9 @@
+# Bare-metal semihosting sample
+
 This sample shows how to use semihosting with
 [QEMU Arm System emulator](https://www.qemu.org/docs/master/system/target-arm.html)
 targeting the
 [micro:bit board model](https://www.qemu.org/2019/05/22/microbit/).
 
 It uses the startup code and the linker script file from the GNU Arm Embedded
 Toolchain samples.
```
Comment (vhscampos) [inline samples/src/baremetal-semihosting/README.md:9]: Consider adding a note to say that qemu 5.2.0 crashes on this sample.

**Review by vhscampos — CHANGES_REQUESTED** [review]

File: samples/src/baremetal-semihosting/README.md (hunk 5 above)
Comment (vhscampos) [reply samples/src/baremetal-semihosting/README.md:9]: I only tested with 5.2.0. Maybe other versions display the same bug.

Comment (miyuki) [reply samples/src/baremetal-semihosting/README.md:9]: I tried 4.2.1 (part Ubuntu 20 distribution) and it freezes. Do we know which version works?

Comment (voltur01) [reply samples/src/baremetal-semihosting/README.md:9]: The original samples worked for me with QEMU 5.0.0 built from source, on Ubuntu 16.04.

Comment (miyuki) [reply samples/src/baremetal-semihosting/README.md:9]: Yes, it looks like the sample is broken. I checked Simon's hello.hex file attached to https://bugs.launchpad.net/qemu/+bug/1918302, it works with the latest QEMU 6, but the current version of the sample doesn't. I'll raise a separate issue to fix the sample.

**Pull request merged**